test/queries/common_table_expression_test.rb

```diff
@@ -61,7 +61,7 @@
 
   describe '.from_cte(common_table_expression_hash)' do
     it 'generates an expression with the CTE as the main table' do
-      query = Person.from_cte('lucky_number_seven', Person.where(lucky_number: 7)).where(id: 5)
+      query = Person.from_cte('lucky_number_seven', Person.where(lucky_number: 7)).where(lucky_number_seven: { id: 5 })
```

Comment:
This is changing the spec to match broken behavior instead of fixing the bug. The `lucky_number_seven` alias is what we should expect to be used by default -- and in fact it is done so in both the SELECT and FROM clauses in the generated query (just not the WHERE) clause.